### PR TITLE
mbedtls 3.1.0 gauche 0.9.11-p1

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
previous effort, #91501

### CODE REVIEW

File: Formula/mbedtls.rb [hunk 1]
```diff
@@ -37,6 +37,7 @@ def install
                     "-DUSE_SHARED_MBEDTLS_LIBRARY=On",
                     "-DPython3_EXECUTABLE=#{which("python3")}",
                     "-DCMAKE_INSTALL_RPATH=#{rpath}",
+                    "-DENABLE_TESTING=OFF",
```
Comment (carlocab) [inline Formula/mbedtls.rb:40]: We shouldn't do this. If the upstream test suite fails that should be reported and not disabled.

File: Formula/librist.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@ class Librist < Formula
   url "https://code.videolan.org/rist/librist/-/archive/v0.2.6/librist-v0.2.6.tar.gz"
   sha256 "88b35b86af1ef3d306f33674f2d9511a27d3ff4ec76f20d3a3b3273b79a4521d"
   license "BSD-2-Clause"
+  revision 1
```
Comment (carlocab) [inline Formula/librist.rb:7]: Have we already seen a linkage or test failure from this formula due to the version bump?

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/mbedtls.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/mbedtls.rb:40]: Yeah, sounds good, let me report this onto the upstream.

File: Formula/librist.rb (hunk 2 above)
Comment (nandahkrishna) [reply Formula/librist.rb:7]: Seems like the linkage failure was seen in [this run](https://github.com/Homebrew/homebrew-core/runs/4651881506?check_suite_focus=true).

File: Formula/mbedtls.rb (hunk 1 above)
Comment (nandahkrishna) [reply Formula/mbedtls.rb]: I've added `ENV.deparallelize` to try out a non-parallel build, as suggested by upstream.

Comment (carlocab) [reply Formula/mbedtls.rb]: Does passing `-DGEN_FILES=OFF` not work?

Comment (nandahkrishna) [reply Formula/mbedtls.rb]: > Pass -DGEN_FILES=OFF to cmake. This should use the generated files from the archive. (And I think it should be the default in the release, both with make and with cmake.)

Based on the comment, I assumed these files would be present in the "final" release archive (https://github.com/ARMmbed/mbedtls/archive/v3.1.0.tar.gz), which isn't identical to the archive we're using (https://github.com/ARMmbed/mbedtls/archive/mbedtls-3.1.0.tar.gz). I'll try this solution locally and update the PR if it works.

Comment (nandahkrishna) [reply Formula/mbedtls.rb]: Ah, the checksum difference is only due to the filenames being different. Using `-DGEN_FILES=OFF` seems to work, so I've updated the PR.

**Comment by nandahkrishna** [comment]
`gauche` linkage failure seen in [this run](https://github.com/Homebrew/homebrew-core/runs/4700205195?check_suite_focus=true), revision bumped.

**Comment by nandahkrishna** [comment]
I've opened a separate PR to fix `gauche`'s `livecheck`, will rebase this and update `gauche` to `0.9.11-p1` after that is merged.

**Review by carlocab — APPROVED** [review]
🎉

**Pull request closed without merging**